tests/test-update-install-flatpaks.c

```diff
@@ -2282,87 +3868,178 @@ test_update_deploy_fail_flatpaks_not_deployed (EosUpdaterFixture *fixture,
+  installed_app_executable = g_file_new_for_path (installed_app_executable_path);
 
-  g_file_make_directory_with_parents (expected_directory, NULL, &error);
+  g_file_make_directory_with_parents (installed_app_executable, NULL, &error);
   g_assert_no_error (error);
 
-  expected_directory_child_str = g_file_get_path (expected_directory_child);
-  g_file_set_contents (expected_directory_child_str, "", -1, &error);
+  /* Now simulate a reboot by running eos-updater-flatpak-installer */
+  deployment_csum = get_checksum_for_deploy_repo_dir (deployment_repo_dir,
+                                                      refspec,
+                                                      &error);
   g_assert_no_error (error);
 
-  /* Attempt to update client - run updater daemon */
-  eos_test_client_run_updater (data->client,
-                               &main_source,
-                               1,
-                               NULL,
-                               &updater_cmd,
-                               NULL);
-  g_assert_no_error (error);
+  eos_test_run_flatpak_installer (data->client->root,
+                                  deployment_csum,
+                                  default_remote_name,
+                                  &error);
+  g_assert_error (error, G_SPAWN_EXIT_ERROR, EUFI_EXIT_CODE_APPLY_FAILED);
+  g_clear_error (&error);
 
-  /* Trigger update */
-  autoupdater_root = g_file_get_child (data->fixture->tmpdir, "autoupdater");
-  autoupdater = eos_test_autoupdater_new (autoupdater_root,
-                                          UPDATE_STEP_APPLY,
-                                          1,  /* interval (days) */
-                                          TRUE,  /* force update */
-                                          &error);
+  /* Assert that our runtime was installed */
+  deployed_flatpaks = eos_test_get_installed_flatpaks (updater_directory, &error);
   g_assert_no_error (error);
 
-  /* Done with update, reap updater server */
-  eos_test_client_reap_updater (data->client,
-                                &updater_cmd,
-                                &reaped_updater,
-                                &error);
-  g_assert_no_error (error);
+  g_assert_true (g_strv_contains ((const gchar * const *) deployed_flatpaks, "org.test.Runtime"));
 
-  /* Now simulate a reboot by running eos-updater-flatpak-installer */
+  /* Assert that the counter was not touched */
+  g_assert_false (g_file_test (updater_state_file_path, G_FILE_TEST_EXISTS));
+}
+
+/* Insert a list of flatpaks to automatically install on the commit,
+ * including a flatpak that has a dependency on a runtime that is not
+ * yet installed. Pull the update, kill the network connection by
+ * deleting the remote ostree repo and then simulate a reboot by running
+ * eos-updater-flatpak-installer installer. Both the incoming package
+ * and the dependency should be installed (in this case, dependency
```

Comment:
Is this comment still correct? The dependencies should already be installed and there is no reliance on a cache.

Reply:
I believe this comment has now bee updated.